### PR TITLE
pkg/lwip: Use NETIF_FOREACH macro in sock implementation

### PR DESCRIPTION
**Pull request by yarrick**
### Contribution description

Use lwIP macro to iterate netifs instead of doing it manually. This reduces
dependency on lwIP internals (for instance it will not break if `LWIP_SINGLE_NETIF` is used)

### Testing procedure

All lwip tests still pass.

### Issues/PRs references

None

### CODE REVIEW

**Comment by yarrick** [comment]
Included netif header to try to fix the cppcheck warning.

File: pkg/lwip/contrib/sock/ip/lwip_sock_ip.c [hunk 1]
```diff
@@ -79,7 +80,8 @@ static uint16_t _ip4_addr_to_netif(const ip4_addr_p_t *addr)
     assert(addr != NULL);
 
     if (!ip4_addr_isany(addr)) {
-        for (struct netif *netif = netif_list; netif != NULL; netif = netif->next) {
+        struct netif *netif;
+        NETIF_FOREACH(netif) {
```
Comment (miri64) [inline pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:84]: Here and everywhere: Please add a cppcheck suppression comment, like suggested by the Github annotation.

**Review by miri64 — CHANGES_REQUESTED** [review]

File: pkg/lwip/contrib/sock/ip/lwip_sock_ip.c (hunk 1 above)
Comment (yarrick) [reply pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:84]: I am not sure why it was accepted in the [ifconfig](https://github.com/RIOT-OS/RIOT/commit/eab317749f82764c6e978ab13a6819b5320a031c#diff-989568003e3c26ef6bf5592983b5243864649c98090cf416e47ace07030bcce3R88) code but not here

Comment (miri64) [reply pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:84]: Not sure either. But `cppcheck` can be flaky like that.

**Comment by yarrick** [comment]
Oops, I guess that should have been a fixup commit instead of a force push.

**Review by miri64 — COMMENTED** [review]
ACK

**Review by miri64 — APPROVED** [review]
Please squash!

**Comment by yarrick** [comment]
Ping

**Pull request merged**